### PR TITLE
build(external-controller): convert to fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
semi-automated conversion

### CODE REVIEW

File: examples/service-clients/azure-client/external-controller/package.json [hunk 1]
```diff
@@ -11,13 +11,26 @@
...
+			"import": {
+				"types": "./lib/src/app.d.ts",
+				"default": "./lib/src/app.js"
+			},
+			"require": {
+				"types": "./dist/src/app.d.ts",
+				"default": "./dist/src/app.js"
+			}
+		}
+	},
```
Comment (DLehenbauer) [inline examples/service-clients/azure-client/external-controller/package.json:26]: Note that main/module/types originally pointed to non-existent files.

I guess it didn't matter as long as webpack's config could find them.

File: examples/service-clients/azure-client/external-controller/package.json [hunk 2]
```diff
@@ -11,13 +11,26 @@
...
+	"main": "dist/src/app.js",
+	"types": "dist/src/app.d.ts",
 	"scripts": {
 		"build": "fluid-build . --task build",
 		"build:compile": "fluid-build . --task compile",
-		"build:esnext": "tsc-multi --config ../../../../common/build/build-common/tsc-multi.esm.json",
+		"build:esnext": "tsc --project ./tsconfig.json",
+		"check:are-the-types-wrong": "attw --pack . --entrypoints .",
```
Comment (DLehenbauer) [inline examples/service-clients/azure-client/external-controller/package.json:33]: 'attw' is how I found the wrong entry points.  I debated whether it was worth it for examples.  I guess it is?

File: examples/service-clients/azure-client/external-controller/src/app.ts [hunk 3]
```diff
@@ -16,9 +16,9 @@ import { InsecureTokenProvider } from "@fluidframework/test-runtime-utils";
 
 import { v4 as uuid } from "uuid";
 
-import { AzureFunctionTokenProvider } from "./AzureFunctionTokenProvider";
-import { DiceRollerController, DiceRollerControllerProps } from "./controller";
-import { makeAppView } from "./view";
+import { AzureFunctionTokenProvider } from "./AzureFunctionTokenProvider.js";
+import { DiceRollerController, DiceRollerControllerProps } from "./controller.js";
+import { makeAppView } from "./view.js";
```
Comment (DLehenbauer) [inline examples/service-clients/azure-client/external-controller/src/app.ts:21]: All changes to *.ts files courtesy of ts2esm.

File: examples/service-clients/azure-client/external-controller/webpack.test.cjs [hunk 4]
```diff
@@ -14,6 +14,9 @@ module.exports = (env) => {
 		},
 		resolve: {
 			extensions: [".ts", ".tsx", ".js"],
+			extensionAlias: {
+				".js": [".ts", ".tsx", ".js", ".cjs", ".mjs"],
+			},
```
Comment (DLehenbauer) [inline examples/service-clients/azure-client/external-controller/webpack.test.cjs:19]: @sonalideshpandemsft - This is something that stumped me for a while.  Adding 'extensionAlias' to 'webpack.config.cjs' seemed to fix things (i.e., it worked in npm run start), but jest/puppeteer still reported not being able to resolve modules.

Turns out the 'webpack.test.cjs' doesn't inherit from the root 'webpack.config.js'.

Comment (sonalideshpandemsft) [reply examples/service-clients/azure-client/external-controller/webpack.test.cjs:19]: does this also fix the attw failures?

File: examples/service-clients/azure-client/external-controller/tsconfig.json [hunk 5]
```diff
@@ -1,12 +1,8 @@
 {
-	"extends": [
-		"../../../../common/build/build-common/tsconfig.base.json",
-		"../../../../common/build/build-common/tsconfig.cjs.json",
-	],
+	"extends": "../../../../common/build/build-common/tsconfig.node16.json",
 	"exclude": ["dist", "node_modules"],
 	"compilerOptions": {
```
Comment (jason-ha) [inline examples/service-clients/azure-client/external-controller/tsconfig.json:4]: exclude "li" as well

**Review by jason-ha — COMMENTED** [review]
This example is an app and not a library; so, one case but not the most common.
I think that most example apps should just use ESM and skip CJS.
Most apps (like this one) should not have exports and thus not use attw.

File: examples/service-clients/azure-client/external-controller/webpack.test.cjs (hunk 4 above)
Comment (DLehenbauer) [reply examples/service-clients/azure-client/external-controller/webpack.test.cjs:19]: It was the incorrect export paths (which have now been removed, per Jason's comment.)

File: examples/service-clients/azure-client/external-controller/tsconfig.json (hunk 5 above)
Comment (DLehenbauer) [reply examples/service-clients/azure-client/external-controller/tsconfig.json:4]: I removed CJS, so now it's just 'lib' only.  Thx.

File: examples/service-clients/azure-client/external-controller/package.json [hunk 6]
```diff
@@ -68,6 +65,7 @@
 		"@types/node": "^18.19.0",
 		"@types/uuid": "^9.0.2",
 		"clean-webpack-plugin": "^4.0.0",
+		"copyfiles": "^2.4.1",
 		"cross-env": "^7.0.3",
```
Comment (jason-ha) [inline examples/service-clients/azure-client/external-controller/package.json:69]: unused

File: examples/service-clients/azure-client/external-controller/tsconfig.cjs.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jason-ha) [inline examples/service-clients/azure-client/external-controller/tsconfig.cjs.json:1]: unused file

**Review by jason-ha — APPROVED** [review]
Clean up the unused stuff and then good.

**Pull request merged**